src/pages/[platform]/build-a-backend/add-aws-services/logging/hub-events/index.mdx

```diff
@@ -0,0 +1,135 @@
+import { getCustomStaticPath } from '@/utils/getCustomStaticPath';
+
+export const meta = {
+  title: 'Listen to log events',
+  description: 'Learn how to listen to Hub events for logging',
+  platforms: [
+    'swift',
+    'android'
+  ]
+};
+
+export const getStaticPaths = async () => {
+  return getCustomStaticPath(meta.platforms);
+};
+
+export function getStaticProps(context) {
+  return {
+    props: {
+      meta
+    }
+  };
+}
+
+The Amplify Logger sends errors that occur when using it through Amplify Hub. To ensure that errors do not occur when logging, log messages should be validated and follow the best security practices. Additionally, you should ensure that log messages do not exceed the [Amazon CloudWatch log event size of 256 KB](https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutLogEvents.html).
+
+You can get logging error events by listening/subscribing to the logging events from the Amplify Hub.
+
+
+    <InlineFilter filters={['android']}>
+<BlockSwitcher>
+<Block name="Java">
+
+```java
+Amplify.Hub.subscribe(HubChannel.LOGGING,
+            hubEvent -> {
+                if (hubEvent.getName().equals(LoggingEventName.WRITE_LOG_FAILURE.toString())) {
+                    Log.i("LOGGING", "Failed to write logs");
+                } else if (hubEvent.getName().equals(LoggingEventName.FLUSH_LOG_FAILURE.toString())){
+                    Log.i("LOGGING", "Failed to flush logs");
+                }
+            }
+        );
+```
+
+</Block>
+<Block name="Kotlin">
+
+```kotlin
+Amplify.Hub.subscribe(
+    HubChannel.LOGGING
+) { hubEvent: HubEvent<*> ->
+    if (hubEvent.name == LoggingEventName.WRITE_LOG_FAILURE.toString()) {
+        Log.i("LOGGING", "Failed to write logs")
+    } else if (hubEvent.name == LoggingEventName.FLUSH_LOG_FAILURE.toString()) {
+        Log.i("LOGGING", "Failed to flush logs")
+    }
+}
+```
+
+</Block>
+<Block name="RxJava">
+
+```java
+RxAmplify.Hub.on(HubChannel.LOGGING)
+.map(HubEvent::getName)
+.subscribe(name -> {
+    if (name.equals(LoggingEventName.WRITE_LOG_FAILURE.toString())) {
+        Log.i("LOGGING", "Failed to write logs");
+        return;
+    } else if (name.equals(LoggingEventName.FLUSH_LOG_FAILURE.toString())) {
+        Log.i("LOGGING", "Failed to flush logs");
+        return;
+    }
+});
+```
+</Block>
+</BlockSwitcher>
+
+    </InlineFilter>
+
+    <InlineFilter filters={['swift']}>
+<BlockSwitcher>
+
+<Block name="Listener">
+
+```swift
+import Amplify
+```
+
+```swift
+// Assumes `unsubscribeToken` is declared as an instance variable in your view
+unsubscribeToken = Amplify.Hub.listen(to: .logging) { payload in
+    switch payload.eventName {
+    case HubPayload.EventName.Logging.writeLogFailure:
+        print("Error writing to local log")
+    case HubPayload.EventName.Logging.flushLogFailure:
+        print("Error sending log events to CloudWatch")
+    default:
+        break
+    }
+}
+```
```

Comment:
Are these two snippets in different codeblocks on purpose? It looks kind of weird to me:

![Screenshot 2024-04-18 at 3 09 08 PM](https://github.com/aws-amplify/docs/assets/54393192/37d2f3e0-d3ea-406a-b15c-a8d65755c9d5)

Reply:
Yes they are...This is based on Gen1 docs for iOS